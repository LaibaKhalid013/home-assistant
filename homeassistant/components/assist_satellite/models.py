"""Models for assist satellite."""

from enum import IntFlag, StrEnum


class AssistSatelliteState(StrEnum):
    """Valid states of an Assist satellite entity."""

    LISTENING_WAKE_WORD = "listening_wake_word"
    """Device is streaming audio for wake word detection to Home Assistant."""

    LISTENING_COMMAND = "listening_command"
    """Device is streaming audio with the voice command to Home Assistant."""

    PROCESSING = "processing"
    """Home Assistant is processing the voice command."""

    RESPONDING = "responding"
    """Device is speaking the response."""


class AssistSatelliteEntityFeature(IntFlag):
    """Supported features of Assist satellite entity."""

    ANNOUNCE = 1
    """Device supports remotely triggered announcements."""
