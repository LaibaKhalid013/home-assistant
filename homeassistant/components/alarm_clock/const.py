"""Constants for the Alarm Clock integration."""

import voluptuous as vol

from homeassistant.const import CONF_ICON, CONF_NAME
import homeassistant.helpers.config_validation as cv

DOMAIN = "alarm_clock"
ENTITY_ID_FORMAT = DOMAIN + ".{}"

ATTR_ALARM_TIME = "alarm_time"
ATTR_REPEAT_DAYS = "repeat_days"
ATTR_NEXT_ALARM = "next_alarm"

CONF_ALARM_TIME = "alarm_time"
CONF_REPEAT_DAYS = "repeat_days"

EVENT_ALARM_CLOCK_STARTED = "alarm_clock.started"
EVENT_ALARM_CLOCK_FINISHED = "alarm_clock.finished"
EVENT_ALARM_CLOCK_CANCELLED = "alarm_clock.cancelled"
EVENT_ALARM_CLOCK_CHANGED = "alarm_clock.changed"

STORAGE_KEY = DOMAIN
STORAGE_VERSION = 1

STORAGE_FIELDS = {
    vol.Required(CONF_NAME): cv.string,
    vol.Optional(CONF_ICON): cv.icon,
    vol.Required(CONF_ALARM_TIME): cv.time,
    vol.Required(CONF_REPEAT_DAYS): cv.weekdays,
}
