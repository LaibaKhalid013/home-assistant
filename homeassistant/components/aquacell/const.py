"""Constants for the Aquacell integration."""

DOMAIN = "aquacell"
DATA_AQUACELL = "DATA_AQUACELL"
