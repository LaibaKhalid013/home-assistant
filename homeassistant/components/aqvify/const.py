"""Constants for the Aqvify integration."""

import logging

DOMAIN = "aqvify"

CONF_DEVICE_KEY = "device_key"

LOGGER = logging.getLogger(__package__)
