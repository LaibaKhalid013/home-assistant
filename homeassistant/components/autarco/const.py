"""Constants for the Autarco integration."""

from __future__ import annotations

from datetime import timedelta
import logging
from typing import Final

DOMAIN: Final = "autarco"
LOGGER = logging.getLogger(__package__)
SCAN_INTERVAL = timedelta(minutes=1)

CONF_PUBLIC_KEY: Final = "public_key"

SENSORS_SOLAR: Final = "solar"
