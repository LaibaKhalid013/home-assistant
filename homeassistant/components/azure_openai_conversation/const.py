"""Constants for the OpenAI Conversation integration."""

import logging

DOMAIN = "azure_openai_conversation"
LOGGER = logging.getLogger(__package__)

CONF_AZURE_OPENAI_RESOURCE = "azure_openai_resource"
CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "gpt-4"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 150
CONF_TOP_P = "top_p"
RECOMMENDED_TOP_P = 1.0
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
AZURE_OPEN_API_VERSION = "2024-02-01"
