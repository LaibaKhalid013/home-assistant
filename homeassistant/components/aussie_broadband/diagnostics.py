"""Provides diagnostics for Aussie Broadband."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

from homeassistant.components.diagnostics import async_redact_data

from .const import DOMAIN

if TYPE_CHECKING:
    from homeassistant.config_entries import ConfigEntry
    from homeassistant.core import HomeAssistant

TO_REDACT = ["address", "ipAddresses", "description", "discounts", "coordinator"]


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, config_entry: ConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    return {
        "services": [
            {
                "service": async_redact_data(service, TO_REDACT),
                "usage": async_redact_data(service["coordinator"].data, ["historical"]),
            }
            for service in hass.data[DOMAIN][config_entry.entry_id]["services"]
        ]
    }
