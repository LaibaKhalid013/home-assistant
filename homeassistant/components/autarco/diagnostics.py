"""Support for the Autarco diagnostics."""

from __future__ import annotations

from typing import Any

from homeassistant.components.diagnostics import async_redact_data
from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
from homeassistant.core import HomeAssistant

from . import AutarcoConfigEntry, AutarcoDataUpdateCoordinator

TO_REDACT = {CONF_EMAIL, CONF_PASSWORD}


async def async_get_config_entry_diagnostics(
    hass: HomeAssistant, config_entry: AutarcoConfigEntry
) -> dict[str, Any]:
    """Return diagnostics for a config entry."""
    autarco_data: list[AutarcoDataUpdateCoordinator] = config_entry.runtime_data

    return {
        "config_entry_data": async_redact_data(dict(config_entry.data), TO_REDACT),
        "sites_data": [
            {
                "id": coordinator.site.site_id,
                "name": coordinator.site.system_name,
                "health": coordinator.site.health,
                "solar": {
                    "power_production": coordinator.data.solar.power_production,
                    "energy_production_today": coordinator.data.solar.energy_production_today,
                    "energy_production_month": coordinator.data.solar.energy_production_month,
                    "energy_production_total": coordinator.data.solar.energy_production_total,
                },
            }
            for coordinator in autarco_data
        ],
    }
