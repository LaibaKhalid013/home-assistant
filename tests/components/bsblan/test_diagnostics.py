"""Tests for the diagnostics data provided by the BSBLan integration."""

import logging
from unittest.mock import AsyncMock, MagicMock, patch

from aiohttp import ClientResponse, ClientSession
from syrupy.assertion import SnapshotAssertion

from homeassistant.components.bsblan import DOMAIN
from homeassistant.components.bsblan.diagnostics import (
    async_get_config_entry_diagnostics,
)
from homeassistant.components.bsblan.models import BSBLanCoordinatorData, BSBLanData
from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
from homeassistant.core import HomeAssistant
from homeassistant.setup import async_setup_component

from tests.common import MockConfigEntry
from tests.typing import ClientSessionGenerator

_LOGGER = logging.getLogger(__name__)


async def test_diagnostics(
    hass: HomeAssistant,
    hass_client: ClientSessionGenerator,
    snapshot: SnapshotAssertion,
) -> None:
    """Test diagnostics."""
    # Create a mock config entry
    config_entry = MockConfigEntry(
        domain=DOMAIN,
        data={
            CONF_HOST: "127.0.0.1",
            CONF_PORT: 80,
            CONF_USERNAME: "admin",
            CONF_PASSWORD: "password",
        },
        entry_id="test_entry_id",
    )

    # Create mock state and sensor data
    mock_state = MagicMock()
    mock_state.to_dict.return_value = {"state": "mocked_state"}
    mock_sensor = MagicMock()
    mock_sensor.to_dict.return_value = {"sensor": "mocked_sensor"}

    # Create mock coordinator data
    mock_coordinator_data = BSBLanCoordinatorData(state=mock_state, sensor=mock_sensor)

    # Create mock coordinator
    mock_coordinator = MagicMock()
    mock_coordinator.data = mock_coordinator_data

    # Create mock device, info, and static data
    mock_device = MagicMock()
    mock_device.to_dict.return_value = {"device": "mocked_device"}
    mock_info = MagicMock()
    mock_info.to_dict.return_value = {"info": "mocked_info"}
    mock_static = MagicMock()
    mock_static.to_dict.return_value = {"static": "mocked_static"}

    # Create mock BSBLAN data
    mock_bsblan_data = BSBLanData(
        coordinator=mock_coordinator,
        client=MagicMock(),
        device=mock_device,
        info=mock_info,
        static=mock_static,
    )

    # Add the config entry to hass
    config_entry.add_to_hass(hass)
    hass.data[DOMAIN] = {config_entry.entry_id: mock_bsblan_data}

    # Set up the BSBLAN integration
    with patch("homeassistant.components.bsblan.async_setup_entry", return_value=True):
        assert await hass.config_entries.async_setup(config_entry.entry_id)
    await hass.async_block_till_done()

    # Ensure the diagnostics component is set up
    assert await async_setup_component(hass, "diagnostics", {})
    await hass.async_block_till_done()

    # Mock the client session and response
    mock_client_session = MagicMock(spec=ClientSession)
    mock_response = MagicMock(spec=ClientResponse)
    mock_response.status = 200
    mock_response.json = AsyncMock(
        return_value={
            "coordinator_data": {
                "state": {"state": "mocked_state"},
                "sensor": {"sensor": "mocked_sensor"},
            },
            "device": {"device": "mocked_device"},
            "info": {"info": "mocked_info"},
            "static": {"static": "mocked_static"},
        }
    )
    mock_client_session.get = AsyncMock(return_value=mock_response)


async def test_diagnostics_to_dict_scenarios(
    hass: HomeAssistant, mock_config_entry: MockConfigEntry
) -> None:
    """Test diagnostics with different to_dict scenarios."""

    class AwaitableResult:
        def __await__(self):
            async def async_result():
                return {"awaitable": "result"}

            return async_result().__await__()

    class MockWithAwaitableToDict:
        def to_dict(self):
            return AwaitableResult()

    class MockWithRegularToDict:
        def to_dict(self):
            return {"regular": "result"}

    class MockWithoutToDict:
        pass

    mock_state = MockWithAwaitableToDict()
    mock_sensor = MockWithRegularToDict()
    mock_device = MockWithoutToDict()

    mock_coordinator_data = BSBLanCoordinatorData(
        state=mock_state,
        sensor=mock_sensor,
    )

    mock_coordinator = MagicMock()
    mock_coordinator.data = mock_coordinator_data

    mock_data = BSBLanData(
        coordinator=mock_coordinator,
        client=MagicMock(),
        device=mock_device,
        info=MockWithAwaitableToDict(),
        static=MockWithRegularToDict(),
    )

    hass.data[DOMAIN] = {mock_config_entry.entry_id: mock_data}

    diagnostics_data = await async_get_config_entry_diagnostics(hass, mock_config_entry)

    expected_data = {
        "info": {"awaitable": "result"},
        "device": mock_device,  # This should be the object itself
        "coordinator_data": {
            "state": {"awaitable": "result"},
            "sensor": {"regular": "result"},
        },
        "static": {"regular": "result"},
    }

    assert diagnostics_data == expected_data


async def test_diagnostics_no_to_dict(
    hass: HomeAssistant, mock_config_entry: MockConfigEntry
) -> None:
    """Test diagnostics when to_dict is callable but returns a mock."""
    mock_state = MagicMock()
    mock_state.to_dict.return_value = "state_string"
    mock_sensor = MagicMock()
    mock_sensor.to_dict.return_value = "sensor_string"

    mock_coordinator_data = BSBLanCoordinatorData(
        state=mock_state,
        sensor=mock_sensor,
    )

    mock_coordinator = MagicMock()
    mock_coordinator.data = mock_coordinator_data

    mock_device = MagicMock()
    mock_device.to_dict.return_value = "device_string"
    mock_info = MagicMock()
    mock_info.to_dict.return_value = "info_string"
    mock_static = MagicMock()
    mock_static.to_dict.return_value = "static_string"

    mock_data = BSBLanData(
        coordinator=mock_coordinator,
        client=MagicMock(),
        device=mock_device,
        info=mock_info,
        static=mock_static,
    )

    hass.data[DOMAIN] = {mock_config_entry.entry_id: mock_data}

    # Mock the format_mac function to return a predictable value
    with patch(
        "homeassistant.helpers.device_registry.format_mac",
        return_value="00:11:22:33:44:55",
    ):
        diagnostics_data = (
            await hass.components.bsblan.diagnostics.async_get_config_entry_diagnostics(
                hass, mock_config_entry
            )
        )

    expected_data = {
        "info": "info_string",
        "device": "device_string",
        "coordinator_data": {
            "state": "state_string",
            "sensor": "sensor_string",
        },
        "static": "static_string",
    }

    assert diagnostics_data == expected_data
