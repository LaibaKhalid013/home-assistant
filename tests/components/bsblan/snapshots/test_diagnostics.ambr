# serializer version: 1
# name: test_diagnostics
  dict({
    'coordinator_data': dict({
      'sensor': dict({
        'sensor': 'mocked_sensor',
      }),
      'state': dict({
        'state': 'mocked_state',
      }),
    }),
    'device': dict({
      'device': 'mocked_device',
    }),
    'info': dict({
      'info': 'mocked_info',
    }),
    'static': dict({
      'static': 'mocked_static',
    }),
  })
# ---
